src/lib/y2storage/filesystems/blk_filesystem.rb

```diff
@@ -121,15 +132,67 @@ def top_level_btrfs_subvolume
         BtrfsSubvolume.new(storage_subvol)
       end
 
-      # Collection of Btrfs subvolumes of the device
+      # The default Btrfs subvolume (typically @).
       #
-      # @return [Array<BtrfsSubvolumes>] empty if it makes no sense for this
-      #   filesystem
-      def btrfs_subvolumes
-        return [] unless supports_btrfs_subvolumes?
+      # @note When a new Btrfs filesystem is created, a top level subvolume is added and
+      #   initialized as default subvolume.
+      #
+      # @see #top_level_btrfs_subvolume
+      #
+      # @return [BtrfsSubvolume, nil] nil if it makes no sense for this filesystem
+      def default_btrfs_subvolume
+        return nil unless supports_btrfs_subvolumes?
 
-        storage_subvols = Storage.to_btrfs(to_storage_value).btrfs_subvolumes
-        storage_subvols.to_a.map { |vol| BtrfsSubvolume.new(vol) }
+        storage_subvol = Storage.to_btrfs(to_storage_value).default_btrfs_subvolume
+        BtrfsSubvolume.new(storage_subvol)
+      end
+
+      # The path that the default btrfs subvolume should have
+      #
+      # @return [String, nil] nil if it makes no sense for this filesystem
+      def default_btrfs_subvolume_path
+        return nil unless supports_btrfs_subvolumes?
+
+        DEFAULT_BTRFS_SUBVOLUME_PATH
+      end
+
+      # Finds a subvolume by its path
+      #
+      # @param path [String] subvolume path
+      # @return [BtrfsSubvolume, nil] nil if it does not find a subvolume with this path
+      def find_btrfs_subvolume_by_path(path)
+        return nil unless supports_btrfs_subvolumes?
+
+        btrfs = Storage.to_btrfs(to_storage_value)
+        storage_subvol = btrfs.find_btrfs_subvolume_by_path(path)
+        BtrfsSubvolume.new(storage_subvol)
+      rescue Storage::BtrfsSubvolumeNotFoundByPath
+        nil
+      end
+
+      # Creates a new default btrfs subvolume if it does not exist or the top level
+      # subvolume is set as default
```

Comment:
On the other hand, from the description it looks like it creates a new subvolume or does nothing, but that's not true. In some cases, it takes an existing subvolume and sets it as default. That's unexpected from the description.